### ISSUE TITLE
Support for encryptedDevice OSDs on partitions

### ISSUE THREAD

**Issue by samcday** [rook/rook#14533]
**Is this a bug report or feature request?**
* Feature Request

**What should the feature do:**

Allow Ceph clusters deployed by Rook to provision and use encryptedDevice OSDs on disk partitions.

**What is use case behind this feature:**

I'm using CoreOS on a fleet of commodity machines, many of these machines have a root disk that is much larger than needed. I'd like to carve out a partition on these devices and give that to Ceph so I can make full use of the available disk capacity.

IIRC this (using partitions as OSDs) was once entirely unsupported in Rook/Ceph, right? Seems that you *can* do this nowadays, but not if you also try to enable the  `encryptedDevice` config option.

**Environment**:

Rook v1.14.9
Ceph version: 18.2.4 reef (stable)

**Comment by travisn** [comment]
IIRC the encrypted OSDs currently require `lvm` mode, while Rook only supports partitions in `raw` mode. @satoru-takeuchi @guits Is that correct? Or what would it take to support this?

**Comment by satoru-takeuchi** [comment]
@travisn 

Correct. Although Ceph itself supports encryted raw mode osd since [this commit](https://github.com/ceph/ceph/commit/159c30d29f009c07e51a4bd7cd5ecea8cf6c8739), Rook only supports encrypted lvm mode osd.

In Ceph, encrypted lvm mode OSD only supports disk and encrypted raw mode OSD would support both disk and partition. [Once I tried to support encryted raw mode OSD in Rook](https://github.com/rook/rook/pull/13169#issue-1981310637), I canceled this work the implementation would take a long time and I didn't have the rook for this work.

Since I still don't have enough time for this work for now, I expect someone else send a PR.

**Comment by samcday** [comment]
I haven't (yet!) made any code contributions to Rook. I'm not sure this would be a good place to start as it seems a little complex and delicate.

Since we've got confirmation now that such a thing is indeed feasible, may we keep this feature request open and hope that a kind (or commercially motivated) soul contributes the necessary patches?

I have short-term plans to grow my cluster further, and also longer term plans that, if successful, would mean a lot of further growth too. At some point I may find I have enough SATA/nvme GBs hiding between the couch cushions that I roll up my own sleeves here ;)

**Comment by samcday** [comment]
Ping to convince the robot overlords this issue should (please) stay open

**Comment by samcday** [comment]
@travisn would you mind swatting away the stale bot here? Pls and thanks :)
